### PR TITLE
On login page, Show password added, replacing url instead of pushing,…

### LINKED ISSUE #41: Login page UI changes and Error Handling
- [x]  The user should not go back to log in on clicking the back button
- [x]  Login page UI
    - [x]  Error handling
    - [x]  Show/Hide Password
    - [x] Loading in questions
    - [x] icon fix (print and export to csv)
    - [x] remove id and shift preview in ques to left

### PR DESCRIPTION
**Pull request by rishigupta1109**
…network error handled

Fixes #41 

image

image

Network error :

image

### CODE REVIEW

**Comment by rishigupta1109** [comment]
fixed the issue.

File: src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -16,15 +16,19 @@ import InputAdornment from "@mui/material/InputAdornment";
...
-  const { setCurrentUser } = useContext(AuthContext);
+  const { currentUser,setCurrentUser } = useContext(AuthContext);
   const navigate = useNavigate();
 
   const [email, setEmail] = useState<string>("");
   const [password, setPassword] = useState<string>("");
   const [loading, setLoading] = useState<boolean>(false);
   const [error, setError] = useState<string>("");
  const [showPassword,setShowPassword]=useState<boolean>(false);
+  if(currentUser!=null){
```
Comment (Anuragtech02) [inline src/pages/Login/Login.tsx:29]: Put this inside a `useEffect` and include `currentUser` in the dependency array.

**Review by Anuragtech02 — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -49,12 +55,17 @@ const Login = () => {
...
-        navigate("/");
+        navigate("/",{ replace: true });
       } else {
+        
       }
     } catch (error: any) {
-      // console.log("True error", error.response);
-      setError(error.response.data.message);
+      console.log("True error", error);
+      if(error&&error.response&&error.response.data){
```
[the remark this answers is not in the document]
Comment (rishigupta1109) [reply src/pages/Login/Login.tsx]: done

File: src/pages/Login/Login.tsx (hunk 1 above)
Comment (rishigupta1109) [reply src/pages/Login/Login.tsx]: done

**Review by Anuragtech02 — APPROVED** [review]
LGTM :)

**Pull request merged**